src/functional/ffront/past_passes/closure_var_type_deduction.py

```diff
@@ -0,0 +1,50 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2022, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either vervisit_Constantsion 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+from typing import Any
+
+import functional.ffront.program_ast as past
+from eve import NodeTranslator, traits
+from functional.ffront.symbol_makers import make_symbol_type_from_value
+
+
+class ClosureVarTypeDeduction(traits.VisitorWithSymbolTableTrait, NodeTranslator):
+    closure_vars: dict[str, Any]
+
+    def __init__(self, closure_vars: dict[str, Any]):
+        self.closure_vars = closure_vars
+
+    @classmethod
+    def apply(cls, node: past.Program, closure_vars: dict[str, Any]):
+        return cls(closure_vars=closure_vars).visit(node)
+
+    def visit_Program(self, node: past.Program, **kwargs):
+        new_closure_vars: list[past.Symbol] = []
+        for sym in node.closure_vars:
+            if sym.id in self.closure_vars and not isinstance(self.closure_vars[sym.id], type):
```

Comment:
```suggestion
            if not isinstance(self.closure_vars[sym.id], type):
```
If `sym.id` is not in `self.closure_vars` this should not silently use the `else` branch, but fail

Reply:
I think some type closure vars were not included at all but still had the symbols for some reason. I removed it and it seems to work now. Nonetheless, the `type` closure vars should still be handled, but I don't know what their type should be. Function (float64) -> float64?

Reply:
Fixed